src/index.js

```diff
@@ -1,94 +1,184 @@
 import F from 'futil'
 import _ from 'lodash/fp'
-import { observable, extendObservable, toJS } from 'mobx'
+import { observable, extendObservable } from 'mobx'
 import * as validators from './validators'
+import { tokenizePath, pickFields, safeJoinPaths } from './util'
+import { treePath, splitAt } from './futil'
+import { get, set, toJS } from './mobx'
+export { validators }
 
+let clone = _.flow(toJS, _.cloneDeep)
 let unmerge = _.flow(F.simpleDiff, _.mapValues('to'))
 let changed = (x, y) => !_.isEqual(x, y) && !(F.isBlank(x) && F.isBlank(y))
 let Command = F.aspects.command(x => y => extendObservable(y, x))
+let throwIfLeaf = x =>
+  !_.has('fields', x) &&
+  !_.has('itemField', x) &&
+  F.throws(`${_.join('.', x.path)}: No fields or itemField`)
+let fieldPath = _.flow(F.intersperse('fields'), _.compact)
 
-let values = _.mapValues('value')
-let Form = ({
+export default ({
   afterInitField = x => x,
   validate = validators.functions,
   ...config
 }) => {
-  let initField = (x, field) => {
+  let saved = {}
+
+  let state = observable({
+    // config.field values are more of default values than anything else
+    // Only supporting them to avoid massive breaking changes
+    value: _.merge(_.mapValues('value', config.fields), config.value),
+    errors: {},
+  })
+
+  let initField = config => {
+    let dotPath = _.join('.', config.path)
+    let valuePath = ['value', ...config.path]
     let node = observable({
-      field,
-      label: x.label || _.startCase(field),
-      // in case no value is provided, avoids controlled vs uncontrolled warning
-      // and need for mobx 4
-      value: _.getOr('', 'value', x),
+      ..._.omit('path', config),
+      field: _.last(config.path),
+      label: config.label || _.startCase(_.last(config.path)),
+      get value() {
+        return get(valuePath, state)
+      },
+      set value(x) {
+        set(valuePath, x, state)
+      },
       get errors() {
-        return form.errors[field] || []
+        return _.isEmpty(dotPath)
+          ? state.errors
+          : get([dotPath], state.errors) || []
       },
       get isValid() {
-        return !node.errors.length
+        return _.isEmpty(node.errors)
       },
       get isDirty() {
-        return changed(toJS(node.value), toJS(x.value))
+        return changed(_.get(valuePath, saved), toJS(node.value))
       },
-      reset() {
-        node.value = F.when(_.isUndefined, '')(x.value)
+      getField(path) {
+        path = fieldPath(tokenizePath(path))
+        return _.isEmpty(path) ? node : _.get(safeJoinPaths(path), node.fields)
       },
-      validate: () => form.validate([field]),
       clean() {
-        x.value = node.value
+        F.setOn(valuePath, clone(node.value), saved)
+      },
+      reset(value) {
+        // Reset errors
+        if (_.isEmpty(node.path)) clearFormErrors()
```

Comment:
these seem like they should be the same code path - clearing errors on a node should traverse its children (e.g. clearing `b` in `{a:{b:{c}}}` should clear `c` also